lib/blockscore/base.rb

```diff
@@ -4,14 +4,28 @@ module BlockScore
   class Base
     extend Connection
 
-    attr_reader :attributes
 
     def initialize(options = {})
+      @loaded = !(options.instance_of?(Proc))
       @attributes = options
     end
 
+    # potential issues here
+    def attributes
+      return @attributes if @loaded
+      force!
+      @attributes
+    end
+
+    def force!
+      @attributes = @attributes.call().attributes
+      @loaded = true
+      self
+    end
+
     def inspect
-      "#<#{self.class}:0x#{object_id.to_s(16)} JSON: " + JSON.pretty_generate(attributes)
+      str_attr = @loaded ? "JSON:#{JSON.pretty_generate(attributes) }": "LAZY:..."
```

Comment:
Style/SpaceAroundOperators: Surrounding space missing for operator ':'.<br>Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.